cityofaustin/atd-vz-data
# Rework requirements.txt and remove `_request_ctx_stack` use from CR3 API service

## Associated issues

This PR is part of closing https://github.com/cityofaustin/atd-data-tech/issues/14729.

## Testing

Testing should be local for this one. Here are some steps to test:

1) create a `.env` in the `atd-cr3-api` folder if you don't have one already. It may be easiest to just ask me to wormhole you one, or you can piece it together out of our 1PW entries. If you happen to have an existing env file, please make sure it has `API_CLIENT_ID` and `API_CLIENT_SECRET`.
1) Spin up the flask app after creating a venv for it and installing the requirements set forth in the `requirements.txt`. This should setup an API endpoint for you at `localhost:3010`, which you can visit to get a status message at the `/` route.
1) Now, head over to your VZE, and edit the environment file for it. You'll want to set the following value as: `REACT_APP_CR3_API_DOMAIN="http://localhost:3010"`
1) Now you can use the VZE to interact with this endpoint. You can read the [flask app's routes](https://github.com/cityofaustin/atd-vz-data/blob/cr3-api-dependabot/atd-cr3-api/server.py) to see what you can test, but here are some highlights:

  * Can you download a CR3?
  * List users on the users page? 
 
---
#### Ship list
